cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -791,7 +800,20 @@ public CloudQuota getQuota(String quotaName, boolean required) {
 
     @Override
     public List<ApplicationLog> getRecentLogs(String applicationName) {
-        throw new UnsupportedOperationException(MESSAGE_FEATURE_IS_NOT_YET_IMPLEMENTED);
+        UUID appGuid = getRequiredApplicationGuid(applicationName);
+        RecentLogsRequest request = RecentLogsRequest.builder()
+                                                     .applicationId(appGuid.toString())
+                                                     .build();
+        return dopplerClient.recentLogs(request)
+                            .map(Envelope::getLogMessage)
+                            .map(log -> new ApplicationLog(log.getApplicationId(),
```

Comment:
Follow the pattern in this class. Create a `RawApplicationLog` class and do the mapping to `ApplicationLog` there.

Reply:
Do I need a "raw" class or just change Application log to an interface and use immutables?